interfaces/builtin/network_control.go

```diff
@@ -65,7 +65,14 @@ dbus (send)
      bus=system
      path="/org/freedesktop/resolve1"
      interface="org.freedesktop.resolve1.Manager"
-     member="SetLink{DNS,MulticastDNS,Domains,LLMNR}"
+     member="Resolve{Address,Hostname,Record,Service}"
```

Comment:
doesn't this kind of duplicate the rule in line 57? the other only mentions the peer by their dbus name, while this one is more explicit about the peer being unconfined

Reply:
Yes, this is a duplicate - we are talking to systemd-resolved that runs unconfined, so we need `peer=(label=unconfined)` too. Denials happen otherwise, although in theory the previous rule should be enough. I am actually wondering if we should remove the previous one, but I am afraid of regressions.

This is the denial I see without the new rule:
```
[ 4058.621805] audit: type=1107 audit(1646825184.452:356): pid=592 uid=100 auid=4294967295 ses=4294967295 subj=? msg='apparmor="DENIED" operation="dbus_method_call"  bus="system" path="/org/freedesktop/resolve1" interface="org.freedesktop.resolve1.Manager" member="ResolveAddress" mask="send" name=":1.0" pid=2615 label="snap.network-manager.networkmanager" peer_pid=555 peer_label="unconfined"
```
cc @alexmurray

Reply:
@alfonsosanchezbeato have you tried just modifying the above line to have:

```
peer=(name="org.freedesktop.resolve1", label="unconfined"),
```

AFAIK it's valid to have both specified in the peer thing see https://gitlab.com/apparmor/apparmor/-/wikis/AppArmorDBus#dbus-rule-syntax

Reply:
Yes, that works, thanks for the hint! I have force-pushed now.